source/extensions/common/tap/tap_matcher.cc

```diff
@@ -152,7 +152,9 @@ HttpGenericBodyMatcher::HttpGenericBodyMatcher(
       NOT_REACHED_GCOVR_EXCL_LINE;
     }
     // overlap_size_ indicates how many bytes from previous data chunk(s) are buffered.
-    overlap_size_ = std::max(overlap_size_, patterns_->back().length() - 1);
+    if (patterns_->back().length() != 0) {
+      overlap_size_ = std::max(overlap_size_, patterns_->back().length() - 1);
```

Comment:
@cpakulski does this line up to expectations for the matcher? Maybe the original intent for `std::max` was to prevent this case.

Reply:
I think that API's verification code would make sure that `patterns_` contains strings of at least 1 character long. But if somehow it contains empty pattern` "",` I agree that it would cause a crash.

Reply:
Ah I see, thanks! `patterns_` is API verified to contain at least one pattern, but the patterns can be empty.
https://github.com/envoyproxy/envoy/blob/49d4a2a54be31c226fc26ee5976007d01b76e2ba/api/envoy/config/tap/v3/common.proto#L121

Based on that, seems like the better way to do this is to verify patterns are non-empty -- @stedsome let's change this to an API validation for non-empty patterns

Reply:
@asraa @stedsome I can add it within 5 minutes. Let me know if you want to continue on that or you want me to open a small PR.

Reply:
@cpakulski I mean for sure it will be quicker for you do this. But if you think this is doable for someone with no prior-knowledge of this( I think it looks straightforward ), I am more happy to try and learn.

Reply:
I'd say go for it @stedsome  -- I can review the PGV notation, should be very simple.